src/Models/Messages/Message.php

```diff
@@ -1,291 +1,475 @@
-    protected $embed;
+    private int $timestamp;
 
-    /** @var ?string MemberID (guildID.userID), Null when sending or receiving webhook messages, just (UserID) if DM Channel. */
-    protected $author_id;
+    /** Null if never edited. */
+    private ?int $edited_timestamp;
 
-    /** @var string */
-    protected $channel_id;
+    private bool $tts;
 
-    /** @var ?string Null if DM Channel. */
-    protected $server_id;
+    private bool $mention_everyone;
 
-    /** @var ?float Null when sending message. */
-    protected $timestamp;
+    /** @var string[] User IDs */
+    private array $mentions;
 
-    /** @var Attachment[] Used for INBOUND messages only. */
-    protected $attachments = [];
+    /** @var string[] Role IDs */
+    private array $mention_roles;
 
-    /** @var bool */
-    protected $everyone_mentioned = false;
+    /** @var Attachment[] */
+    private array $attachments;
 
-    /** @var string[] */
-    protected $users_mentioned = [];
+    /** @var Embed[] */
+    private array $embeds;
 
-    /** @var string[] */
-    protected $roles_mentioned = [];
+    /** @var Reaction[] */
+    private array $reactions;
 
-    /** @var string[] */
-    protected $channels_mentioned = [];
+    private bool $pinned;
+
+    private ?string $webhook_id;
+
+    private ?Activity $activity;
+
+    private ?string $application_id;
+
+    private ?Reference $message_reference;
 
     /**
-     * Message constructor.
-     *
-     * @param string       $channel_id
-     * @param string|null  $id
-     * @param string       $content
-     * @param Embed|null   $embed
-     * @param string|null  $author_id
-     * @param string|null  $server_id
-     * @param float|null   $timestamp
-     * @param Attachment[] $attachments
-     * @param bool         $everyone_mentioned
-     * @param string[]     $users_mentioned
-     * @param string[]     $roles_mentioned
-     * @param string[]     $channels_mentioned
+     * @var int|null Flags bitmask.
+     * @see Message::FLAGS
      */
-    public function __construct(string $channel_id, ?string $id = null, string $content = "", ?Embed $embed = null,
-                                ?string $author_id = null, ?string $server_id = null, ?float $timestamp = null,
-                                array $attachments = [], bool $everyone_mentioned = false, array $users_mentioned = [],
-                                array $roles_mentioned = [], array $channels_mentioned = []){
-        $this->setChannelId($channel_id);
+    private ?int $flags;
+
+    /**
+     * This field is only returned for messages with a type of 19 (REPLY) or 21 (THREAD_STARTER_MESSAGE).
+     * If the message is a reply but the referenced_message field is not present,
+     * the backend did not attempt to fetch the message that was being replied to, so its state is unknown.
+     * If the field exists but is null, the referenced message was deleted.
+     */
+    private ?Message $referenced_message;
+
+    //interaction?
```

Comment:
?

Reply:
https://discord.com/developers/docs/interactions/receiving-and-responding#message-interaction-object